spec/unit/promise_spec.rb

```diff
@@ -398,12 +398,20 @@
     end
 
     describe '#fulfill' do
-      it 'does not return anything' do
-        expect(subject.fulfill(nil)).to eq(nil)
+      it 'returns itself to allow chaining with passed nothing' do
+        expect(subject.fulfill).to be(subject)
       end
 
-      it 'does not return anything when given a promise' do
-        expect(subject.fulfill(Promise.new)).to eq(nil)
+      it 'returns itself to allow chaining with passed nil' do
+        expect(subject.fulfill(nil)).to be(subject)
+      end
+
+      it 'returns itself to allow chaining with passed value' do
+        expect(subject.fulfill('value')).to be(subject)
+      end
+
+      it 'returns itself even when passing promise' do
+        expect(subject.fulfill(Promise.new)).to be(subject)
```

Comment:
yeah, but could return passed promise to fulfill instead, and it documents this choice (it turns out it's semantically equvalent, but it wan't that obvious at first)